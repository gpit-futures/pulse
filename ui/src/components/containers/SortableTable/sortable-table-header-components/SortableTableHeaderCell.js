import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const SortableTableHeaderCell = ({ name, title, icon, onClick, sortingOrder }) => (
  <th
    className={classNames({ 'sorted': sortingOrder }, sortingOrder)}
    name={name}
    onClick={e => onClick(e, name)}
  >
    <span>{title}</span> {icon}
  </th>
);

SortableTableHeaderCell.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  sortingOrder: PropTypes.oneOf([null, 'asc', 'desc']),
  icon: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
};

export default SortableTableHeaderCell;
