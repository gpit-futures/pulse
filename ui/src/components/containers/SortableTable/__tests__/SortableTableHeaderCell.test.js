import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import SortableTableHeaderCell from '../sortable-table-header-components/SortableTableHeaderCell';

Enzyme.configure({ adapter: new Adapter() });

const onClick = () => console.log('test');

describe('Component <SortableTableHeaderCell />', () => {
  it('should renders correctly', () => {
    const sortableTableHeaderCell = shallow(
      <SortableTableHeaderCell
        name="cause"
        sortingOrder="asc"
        title="Cause"
        onClick={onClick}
      />);
    expect(sortableTableHeaderCell).toMatchSnapshot();
    sortableTableHeaderCell.find('.asc').simulate('click');
  });
});

