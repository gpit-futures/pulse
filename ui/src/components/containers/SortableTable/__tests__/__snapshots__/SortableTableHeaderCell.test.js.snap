// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <SortableTableHeaderCell /> should renders correctly 1`] = `
<th
  className="sorted asc"
  name="cause"
  onClick={[Function]}
>
  <span>
    Cause
  </span>
   
</th>
`;
